### PR TITLE
chore: exit if unhandledRejection during `TEST_MODE`

### PR DESCRIPTION
**Pull request by ggazzo**
[ARCH-1308](https://rocketchat.atlassian.net/browse/ARCH-1308)

after the first run:

image
I can conclude we have a few unhandled cases under development which is bad :s

## Further comments

[ARCH-1308]: https://rocketchat.atlassian.net/browse/ARCH-1308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by KevLehman — APPROVED** [review]

File: apps/meteor/server/email/IMAPInterceptor.ts [hunk 1]
```diff
@@ -121,26 +121,19 @@ export class IMAPInterceptor extends EventEmitter {
...
 			this.backoffDurationMS = 3000;
 		}
-		const loop = async (): Promise<void> => {
-			logger.debug(`Reconnecting to ${this.config.user}: ${this.retries}`);
-			if (this.canRetry()) {
-				this.backoffDurationMS *= 2;
-				this.backoff = setTimeout(loop, this.backoffDurationMS);
-			} else {
-				logger.info(`IMAP reconnection failed on inbox ${this.config.user}`);
-				clearTimeout(this.backoff);
```
Comment (debdutdeb) [inline apps/meteor/server/email/IMAPInterceptor.ts:135]: these seem unrelated?

Comment (ggazzo) [reply apps/meteor/server/email/IMAPInterceptor.ts:135]: it depends, its related because without that this pr will never pass since we dont handle imap errors.

but its not related because involve fixes, refactors whatever... because of that cherry-picked the commit
https://github.com/RocketChat/Rocket.Chat/pull/33855

as soon this one get merged, i will rebase and this changes will be dropped

**Review by jeanfbrito — APPROVED** [review]
